File: Services/MouseDeviceFiltering.vb
Imports WpfApp1.Models

Namespace Services
    Friend Module MouseDeviceFiltering
        Public Function FilterPhysicalDevices(devices As IEnumerable(Of RawMouseDeviceInfo)) As IReadOnlyList(Of RawMouseDeviceInfo)
            If devices Is Nothing Then
                Return Array.Empty(Of RawMouseDeviceInfo)()
            End If

            Return devices.
                Where(Function(device) device IsNot Nothing AndAlso Not device.IsVirtual).
                ToArray()
        End Function
    End Module
End Namespace

File: src/ClickSyncMouseTester/Services/MouseDeviceFiltering.cs
using ClickSyncMouseTester.Models;
using System;
using System.Collections.Generic;

namespace ClickSyncMouseTester.Services;

internal sealed class MouseDeviceFiltering
{
    private static readonly RawMouseDeviceCatalog Catalog = new RawMouseDeviceCatalog();

    public static IReadOnlyList<RawMouseDeviceInfo> FilterSelectableMotionDevices(IEnumerable<RawMouseDeviceInfo> devices, IReadOnlyDictionary<string, RawMouseEndpointActivitySnapshot> activitySnapshots)
    {
        if (devices == null)
        {
            return Array.Empty<RawMouseDeviceInfo>();
        }

        return Catalog.CreateSelectableDevices(devices, activitySnapshots);
    }

    public static bool ContainsSelectableMotionDevice(IEnumerable<RawMouseDeviceInfo> devices, IReadOnlyDictionary<string, RawMouseEndpointActivitySnapshot> activitySnapshots, string deviceId)
    {
        return Catalog.ContainsSelectableDevice(devices, activitySnapshots, deviceId);
    }
}





